### PR TITLE
General: updated minimum required WP version to run Jetpack

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

* This PR does 2 things:
1. It updates the minimum required WP version to run Jetpack, so we can remain at WP-1 when WordPress 5.3 becomes available.
2. It removes the checks we had installed to check the installed version of PHP and pause Jetpack, since WordPress 5.2 requires PHP 5.6, which is also what Jetpack requires.

**Notes**

* I am not sure we should ship this in Jetpack 7.9, since when we release it WordPress 5.3 won't be available yet. Maybe we should wait until Jetpack 8.0?
* I'm not attached to 7955bdad406a5ac93b2ebead469f125a7f5a9f9b. I'd be happy to leave it in, just in case Jetpack decides to move away from Core's PHP requirements once again. I'm happy to revert it if needed.

#### Testing instructions:

* Start on a site running WordPress 5.1
* Add this patch
* Jetpack will be paused

image

#### Proposed changelog entry for your changes:

* General: As WordPress 5.3 is now available, Jetpack now requires WordPress 5.2.

### CODE REVIEW

**Comment by jeherve** [comment]
> I am not sure we should ship this in Jetpack 7.9, since when we release it WordPress 5.3 won't be available yet. Maybe we should wait until Jetpack 8.0?

The Unit Tests agree with this, since they currently fail for the previous version of WP (5.1). I'll change the milestone to 8.0.

**Comment by kraftbj** [comment]
If we really wanted to, we can force Travis into listening to us, e.g. https://github.com/Automattic/jetpack/commit/8e0d2cde08dfa0065dd457bbe43229268b384af1#diff-354f30a63fb0907d4ad57269548329e3

While we're touching things, I don't remember if there was a reason we said `5.6` for minimum PHP vs `5.6.20` which is WordPress' minimum now.

**Comment by jeherve** [comment]
> I don't remember if there was a reason we said 5.6 for minimum PHP vs 5.6.20 which is WordPress' minimum now.

I don't think there was.

> If we really wanted to, we can force Travis into listening to us

Let's wait one month, I think that's fine. It gives folks a bit more time to upgrade.

**Review by kraftbj — DISMISSED** [review]
We'll need to wait until 5.3 ships (Nov 12 ETA) before re-running Travis to actually merge.

**Review by kraftbj — APPROVED** [review]

**Comment by jeherve** [comment]
On hold for now; I'll re-run the tests and merge when WordPress 5.3 is released.

**Comment by kraftbj** [comment]
5.3 is out, reran all tests and Travis likes.

File: jetpack.php
Comment (lezama) [inline jetpack.php]: 😍

**Pull request merged**
